patchwork-biomes/src/main/java/net/minecraftforge/common/BiomeManager.java

```diff
@@ -64,14 +70,134 @@ private OverworldClimate getClimate() {
 	public static class BiomeEntry extends WeightedPicker.Entry {
 		public final Biome biome;
 
+		public int field_76292_a; // evil hack
+
 		public BiomeEntry(Biome biome, int weight) {
 			super(weight);
 
+			this.field_76292_a = weight;
 			this.biome = biome;
 		}
 
 		private int getWeight() {
 			return this.weight;
 		}
 	}
+
+	// Biomes O' Plenty pokes Forge internals. Fun
+	private static TrackedList<BiomeEntry>[] biomes = setupBiomes();
+
+	private static TrackedList<BiomeEntry>[] setupBiomes() {
+		@SuppressWarnings("unchecked")
+		TrackedList<BiomeEntry>[] currentBiomes = new TrackedList[BiomeType.values().length];
+		List<BiomeEntry> list = new ArrayList<BiomeEntry>();
+
+		list.add(new BiomeEntry(Biomes.FOREST, 10));
+		list.add(new BiomeEntry(Biomes.DARK_FOREST, 10));
+		list.add(new BiomeEntry(Biomes.MOUNTAINS, 10));
+		list.add(new BiomeEntry(Biomes.PLAINS, 10));
+		list.add(new BiomeEntry(Biomes.BIRCH_FOREST, 10));
+		list.add(new BiomeEntry(Biomes.SWAMP, 10));
+
+		currentBiomes[BiomeType.WARM.ordinal()] = new TrackedList<BiomeEntry>(list);
+		list.clear();
+
+		list.add(new BiomeEntry(Biomes.FOREST, 10));
+		list.add(new BiomeEntry(Biomes.MOUNTAINS, 10));
+		list.add(new BiomeEntry(Biomes.TAIGA, 10));
+		list.add(new BiomeEntry(Biomes.PLAINS, 10));
+
+		currentBiomes[BiomeType.COOL.ordinal()] = new TrackedList<BiomeEntry>(list);
+		list.clear();
+
+		list.add(new BiomeEntry(Biomes.SNOWY_TUNDRA, 30));
+		list.add(new BiomeEntry(Biomes.SNOWY_TAIGA, 10));
+
+		currentBiomes[BiomeType.ICY.ordinal()] = new TrackedList<BiomeEntry>(list);
+		list.clear();
+
+		currentBiomes[BiomeType.DESERT.ordinal()] = new TrackedList<BiomeEntry>(list);
+
+		return currentBiomes;
+	}
+
+	public static ImmutableList<BiomeEntry> getBiomes(BiomeType type) {
+		int idx = type.ordinal();
+		List<BiomeEntry> list = idx >= biomes.length ? null : biomes[idx];
```

Comment:
this line makes sense but it's difficult to tell what it's doing at first glance
```suggestion
		// We want to return null instead of throw an exception if the index is out of bounds.
		List<BiomeEntry> list = idx >= biomes.length ? null : biomes[idx];
```